src/array/ops/cast.rs

```diff
@@ -701,12 +709,78 @@ impl EmbeddingArray {
 
 impl ImageArray {
     pub fn cast(&self, dtype: &DataType) -> DaftResult<Series> {
-        self.physical.cast(dtype)
+        match dtype {
+            #[cfg(feature = "python")]
+            DataType::Python => Python::with_gil(|py| {
+                let mut ndarrays = Vec::with_capacity(self.len());
+                let da = self.data_array();
+                let ca = self.channel_array();
+                let ha = self.height_array();
+                let wa = self.width_array();
+                let pyarrow = py.import("pyarrow")?;
+                for (i, arrow_array) in da.iter().enumerate() {
+                    let shape = (
+                        ha.value(i) as usize,
+                        wa.value(i) as usize,
+                        ca.value(i) as usize,
+                    );
+                    let py_array = match arrow_array {
+                        Some(arrow_array) => ffi::to_py_array(arrow_array, py, pyarrow)?
+                            .call_method0(py, pyo3::intern!(py, "to_numpy"))?
```

Comment:
Hmm I left it off since it should always be zero-copy right now, since the Arrow array for each image should not contain any nulls and should always be uint8, but I suppose I can add it in without any harm.